system-tests/test.py

```diff
@@ -19,118 +20,124 @@
 import subprocess
 import sys
 import os
-import traceback
-from typing import List, Sequence, Dict
+from typing import Dict, List, Optional, Sequence
 
 import systest  # type: ignore
 from systest import Sequencer
 
 
-class RunTestCase(systest.TestCase):
-    '''
+class TestParameters(object):
+    """
+    Holds information about the environment a test is happening in.
+    """
+    def __init__(self, test_directory: Path, tag: str):
+        self._test_dir = test_directory
+        self._tag = tag
+
+    @property
+    def test_directory(self):
+        return self._test_dir
+
+    @property
+    def work_directory(self):
+        return self._test_dir / 'working'
+
+    @property
+    def tag(self):
+        return self._tag
+
+
+class RunCommand(ABC):
+    """
     Base class for tests containing useful utility functions.
-    '''
-
-    def run_command_check_output(self,
-                                 command: List[str],
-                                 environment: Dict,
-                                 expected: str):
-        '''
-        Run ``command`` in the given ``environment`` and then compare
-        its stdout to some ``expected`` output, raising an exception
-        if either the command fails or the output disagrees.
-        '''
-        thread: subprocess.Popen = subprocess.Popen(command,
-                                                    env=environment,
+    """
+    def __init__(self,
+                 parameters: TestParameters,
+                 command: List[str],
+                 environment: Dict):
+        self._parameters = parameters
+        self._command = command
+        self._environment = environment
+        self._debug_output: Optional[List[str]] = None
+
+        self.return_code: Optional[bool] = None
+        self.standard_out: Optional[str] = None
+        self.standard_error: Optional[str] = None
+
+    @property
+    def test_parameters(self) -> TestParameters:
+        return self._parameters
+
+    @abstractmethod
+    def description(self) -> str:
+        raise NotImplementedError("Abstract methods must be implemented.")
```

Comment:
I think the wording of the `NotImplementedError` here is inconsistent with one further back in the changelog (and potentially with some of the ones elsewhere in the codebase - minor thing but can you check and make sure they keep to the same text?)

Reply:
This was right, the repository was wrong.